packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -32,16 +33,41 @@ typedef GestureMultiTapCallback = void Function(int pointer);
 /// [GestureMultiTapDownCallback] will not end up causing a tap.
 typedef GestureMultiTapCancelCallback = void Function(int pointer);
 
+/// CountdownZoned tracks whether the specified duration has elapsed since
+/// creation, honoring [Zone].
+class _CountdownZoned {
+   _CountdownZoned({ @required Duration duration }) :
+      assert(duration != null) {
+    _timer = Timer(duration, _onTimeout);
+  }
+
+  bool _timeout = false;
+  Timer _timer;
+
+  bool get timeout => _timeout;
+
+  void _onTimeout() {
+    _timeout = true;
+  }
+}
+
 /// TapTracker helps track individual tap sequences as part of a
 /// larger gesture.
 class _TapTracker {
-  _TapTracker({ PointerDownEvent event, this.entry })
-    : pointer = event.pointer,
-      _initialPosition = event.position;
+  _TapTracker({
+    PointerDownEvent event,
+    this.entry,
+    @required Duration doubleTapMinTime,
+  })
+    : assert(doubleTapMinTime != null),
```

Comment:
we should assert that `event != null`.  We should also try to add a message to the assert about why we expect this.

Reply:
And then we should mark `event` as `@required`.

Reply:
I see a lot of places in the codebase using `@require` along with `assert(foo != null)` simply to indicate non-optional, and this is a similar case.

Reply:
Yes, the linter will complain at you if you assert an optional parameter != null and don't mark it as `@required`

Reply:
Yeah, I mean I wonder the necessity of adding assert message here since it's a common case.
I'll add `require` to event later.